### PR TITLE
feat(adaptive-metrics): add segments and exemptions commands

### PR DESCRIPTION
**Pull request by flxbk**
## Summary

- Add full CRUD commands for Adaptive Metrics **segments** (`list`, `get`, `create`, `update`, `delete`) and **exemptions** (`list`, `get`, `create`, `update`, `delete`) under `gcx metrics adaptive`
- Register Segment and Exemption as typed resources with resource adapters, enabling `resources get/push/pull` support
- Add HTTP client methods for the segments (`/aggregations/rules/segments`) and exemptions (`/v1/recommendations/exemptions`) APIs, including cross-segment listing via `--all-segments`
- Unify `doRequest`/`doMutatingRequest` into a single `doRequest(ctx, method, path, body)` matching the adaptive-logs client pattern

## Details

**New leaf commands** (10 total):
- `gcx metrics adaptive segments {list,get,create,update,delete}`
- `gcx metrics adaptive exemptions {list,get,create,update,delete}`

**Features**:
- Table, wide, JSON, YAML output formats with `--json` field discovery/selection
- Merge-update semantics (GET → merge flags → PUT) for both segments and exemptions
- `--all-segments` flag on exemptions list for cross-segment view
- `--limit` flag on segments and exemptions list
- Confirmation prompts on delete with `--yes` to skip
- Agent-mode annotations for all new commands
- Resource adapter registrations with schema, examples, and natural key for exemptions

## Test plan

- [x] Full CRUD lifecycle tested against live dev environment (`cortexdev05`)
- [x] All output formats verified (table, wide, json, yaml, `--json` field selection)
- [x] Error handling / edge cases tested (missing args, conflicting flags, nonexistent IDs, no-op updates)
- [x] Test data cleaned up after testing
- [x] Unit tests for all segment and exemption client methods (34 new table-driven test cases)
- [x] Sentinel error wrapping verified (`ErrSegmentNotFound`, `ErrExemptionNotFound` wrap `adapter.ErrNotFound`)
- [x] CI passes

### CODE REVIEW

File: internal/providers/metrics/adaptive/commands.go [hunk 1]
```diff
@@ -1032,3 +1067,874 @@ func sortRecommendations(recs []MetricRecommendation, by string, reverse bool) {
...
+					for j := range entries[i].Exemptions {
+						normalizeExemption(&entries[i].Exemptions[j])
+					}
+					total += len(entries[i].Exemptions)
+				}
+				fmt.Fprintf(cmd.ErrOrStderr(), "%d exemption(s) across %d segment(s)\n", total, len(entries))
+				if total == 0 {
+					return nil
+				}
+				return opts.Encode(cmd.OutOrStdout(), entries)
```
[the remark this answers is not in the document]
Comment (flxbk) [reply internal/providers/metrics/adaptive/commands.go:1512]: Fixed in 8b6c5fc — added an explicit error when `--limit` is combined with `--all-segments`.

File: internal/providers/metrics/adaptive/commands.go [hunk 2]
```diff
@@ -1032,3 +1067,874 @@ func sortRecommendations(recs []MetricRecommendation, by string, reverse bool) {
...
+			if err != nil {
+				return err
+			}
+
+			total := len(segments)
+			if opts.Limit > 0 && opts.Limit < total {
+				segments = segments[:opts.Limit]
+			}
+
+			fmt.Fprintf(cmd.ErrOrStderr(), "%d segment(s)\n", total)
```
[the remark this answers is not in the document]
Comment (flxbk) [reply internal/providers/metrics/adaptive/commands.go]: Fixed in 8b6c5fc — now shows "%d of %d segment(s)" when `--limit` truncates, matching the recommendations pattern.

File: internal/providers/metrics/adaptive/commands.go [hunk 3]
```diff
@@ -1032,3 +1067,874 @@ func sortRecommendations(recs []MetricRecommendation, by string, reverse bool) {
...
+
+			for i := range exemptions {
+				normalizeExemption(&exemptions[i])
+			}
+
+			total := len(exemptions)
+			if opts.Limit > 0 && opts.Limit < total {
+				exemptions = exemptions[:opts.Limit]
+			}
+
```
[the remark this answers is not in the document]
Comment (flxbk) [reply internal/providers/metrics/adaptive/commands.go]: Fixed in 8b6c5fc — same fix applied for exemptions list.

**Comment by wardbekker** [comment]
## Shepherd Status (iteration 1)

### CI Checks
| Check | Status |
|-------|--------|
| Documentation | PASS |
| Linters | PASS |
| Tests | PASS |
| zizmor | PASS |
| TruffleHog Secret Scan | PASS |
| Socket Security | PASS |

### Actions Taken
- Rebased onto latest `main` (branch was behind), resolved vendor drift, pushed — all CI green

### Review Comments
All 3 inline comments from `claude[bot]` have been addressed by the PR author in commit `8b6c5fc`:
1. `--limit` + `--all-segments` conflict → added explicit error
2. Segments list truncation message → shows "N of M segment(s)"
3. Exemptions list truncation message → shows "N of M exemption(s)"

### Status
- CI: All green
- Merge state: MERGEABLE, branch up to date
- Review: **REVIEW_REQUIRED** — awaiting human reviewer approval

---
*Updated by shepherd-pr*

**Review by wardbekker — APPROVED** [review]
